### ISSUE TITLE
nan_to_num with bool arrays (Trac #1327)

### ISSUE THREAD

**Issue by thouis** [numpy/numpy#1925]
_Original ticket http://projects.scipy.org/numpy/ticket/1327 on 2009-12-11 by trac user Nicolas Rougier, assigned to unknown._

Using both numpy 1.3.0 and 1.4.0rc1 I got the following exception using nan_to_num on a bool array.

```
>>> import numpy
>>> Z = numpy.zeros((3,3),dtype=bool)
>>> numpy.nan_to_num(Z)
Traceback (most recent call last):
 File "<stdin>", line 1, in <module>
 File "/usr/lib/python2.6/dist-packages/numpy/lib/type_check.py", line
374, in nan_to_num
   maxf, minf = _getmaxmin(y.dtype.type)
 File "/usr/lib/python2.6/dist-packages/numpy/lib/type_check.py", line
307, in _getmaxmin
   f = getlimits.finfo(t)
 File "/usr/lib/python2.6/dist-packages/numpy/core/getlimits.py", line
103, in __new__
   raise ValueError, "data type %r not inexact" % (dtype)
ValueError: data type <type 'numpy.bool_'> not inexact
```

**Comment by gfyoung** [comment]
@charris : I don't know if this patch ever got merged, but I think this issue has been fixed now.  I'm not seeing that exception that was reported, and I'm getting the expected values instead.

**Comment by eric-wieser** [comment]
Yep, this was fixed in #5742
